### PR TITLE
ENH: Added ThinImage Python script

### PR DESCRIPTION
**Pull request by kian-weimer**
Created a Python ThinImage example based off of the existing C++ example.

### CODE REVIEW

**Review by dzenanz — APPROVED** [review]
Awesome!

File: src/Filtering/BinaryMathematicalMorphology/ThinImage/Code.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    image.Allocate()
+    image.FillBuffer(0)
+
+    # Draw a 5 pixel wide line
+    for i in range(20, 80):
+        for j in range(50, 55):
+            index = itk.Index[Dimension]()
+            index[0] = i
+            index[1] = j
+            image.SetPixel(index, 255)
```
Comment (thewtex) [inline src/Filtering/BinaryMathematicalMorphology/ThinImage/Code.py:35]: We can reduce this to:

```
image[20:80, 50:55] = 255
```

File: src/Filtering/BinaryMathematicalMorphology/ThinImage/Code.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    # Write Image
+    itk.imwrite(image, "input.png")
+
+image = itk.binary_thinning_image_filter(image)
+
+# Rescale the image so that it can be seen (the output is 0 and 1, we want 0 and 255)
+image = itk.rescale_intensity_image_filter(image, OutputMinimum=0, OutputMaximum=255)
```
Comment (thewtex) [inline src/Filtering/BinaryMathematicalMorphology/ThinImage/Code.py:43]: For a more Pythonic style, `OutputMinimum` -> `output_minimum` and `OutputMaximum` -> `output_maximum`.

**Review by thewtex — CHANGES_REQUESTED** [review]
@kian-weimer thanks for the contribution!

A few style issues are noted inline.

File: src/Filtering/BinaryMathematicalMorphology/ThinImage/Code.py (hunk 1 above)
Comment (kian-weimer) [reply src/Filtering/BinaryMathematicalMorphology/ThinImage/Code.py]: I had to swap the 20:80 and 50:55 to get the line going the correct way, but then it worked, thank you.

File: src/Filtering/BinaryMathematicalMorphology/ThinImage/Code.py (hunk 2 above)
Comment (kian-weimer) [reply src/Filtering/BinaryMathematicalMorphology/ThinImage/Code.py]: Done

**Review by dzenanz — APPROVED** [review]

**Pull request merged**